models/analytics/facts/meeting_form_responses_fact.sql

```diff
@@ -0,0 +1,47 @@
+WITH cleaned_encounters AS (
+	SELECT
+		id,
+		CAST(CAST(observations AS JSONB) ->> 'Date of WIMC meeting' AS DATE) AS meeting_date,
```

Comment:
hmm i see. two points 
- Data Freshness
- Storage

wouldn't we take a hit on freshness if we put an intermediate unnesting step, because now this would have to be run every time we process a batch + Additional storage being used and maintenance overhead.

just wondering what's the use of doing flattening / unnesting in the intermediate stage here versus. unnesting on the fly ~ directly on the source data as it arrives.

velocity here would determine which way to go - i don't mind either. lmk what you think

Reply:
Great point. let's discuss this further. 
On **data freshness** - So it depends on whether process incremental data in the intermediate step and then the final model does not have to do a full refresh it will only query and process the new records. So by separating it out we can reduce the amount of compute we are spending on processing all records for each step. 

On **storage** - we can materialize the intermediate step as a view instead so that we don't spend storage costs on this. 

This way we get both best practices as well reduction in costs by saving on compute(process less records in final step) and storage.

Reply:
Got it, cool. I am aligned. Will make necessary updates.